shimmingtoolbox/shim/sequencer.py

```diff
@@ -833,26 +834,33 @@ def select_optimizer(method, unshimmed, affine, coils: ListCoil, pmu: PmuResp =
     return optimizer
 
 
-_optimize_scope = {}  # i?: (_optimizer: Optimizer, nii_mask_anat, slices_anat, shimwise_bounds=None,
-                      #     dilation_kernel='sphere', dilation_size=3, path_output=None) 
-def _optimize(_optimizer: Optimizer, nii_mask_anat, slices_anat, shimwise_bounds=None,
-              dilation_kernel='sphere', dilation_size=3, path_output=None):
+# Used as a global dictionary to pass argument to the _opt function. This avoids to pickle the data resulting in a slow
+# start of the multiprocessing pool
+# uuid: (_optimizer: Optimizer, nii_mask_anat, slices_anat, shimwise_bounds=None, dilation_kernel='sphere',
+# dilation_size=3, path_output=None)
+_optimize_scope = {}
 
+
+def _optimize(optimizer: Optimizer, nii_mask_anat, slices_anat, shimwise_bounds=None,
+              dilation_kernel='sphere', dilation_size=3, path_output=None):
     # Count shims to perform
     n_shims = len(slices_anat)
-    
-    global _optimize_scope
-    global optimizer
-    optimizer = _optimizer # not a super safe thing to do
 
     # multiprocessing optimization
     mp.set_start_method('fork', force=True)
```

Comment:
Since this sets a global value (inside mp, but still global) maybe it should be done right after the `import`s, or maybe in the script's `main`.

Reply:
I think we should put it at the top of this file, the reason I say this is that the script is not the only thing calling this API, other tests are as well. Otherwise we need to also set it in every test that has mp (sequencer). But maybe that's not too bad.. What do you think?

Reply:
For what it's worth, python [suggests](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods)

> To select a start method you use the set_start_method() in the if __name__ == '__main__' clause of the main module.

But since the default on Linux is 'fork', the default on macOS is 'spawn' like you [suggested](https://github.com/shimming-toolbox/shimming-toolbox/pull/390#issuecomment-1130515864) so maybe we should just not set it at all, but then we're back to square one, and it'll just be fast on Linux and slow on macOS.